minio/credentials/aws_iam.py

```diff
@@ -42,32 +42,31 @@ def request_cred_list(self):
         url = self._endpoint + self.iam_security_creds_path
         try:
             res = self._http_client.urlopen('GET', url)
-            if res['status'] != 200:
-                return []
+            if res.status != 200:
+                return None
         except:
-            return []
-        creds = res['data'].split('\n')
-        return creds
+            return None
+        creds = res.data
```

Comment:
Could we trim any '\n' or white space that we can find ? I guess that old logic of the code is there for a reason.

Reply:
The `split(\n)` was used in case more than one role name is returned  by AWS metadata service. However, according to AWS docs an EC2 machine can only have one role attached.

I added the `split(\n)` back just in case!